### PR TITLE
Remove ~/ handling from core Razor.

### PR DESCRIPTION
**Pull request by NTaylorMullen**
- Removed parsing, chunk generation and code generation.
- Removed related tests.
- Did not modify existing unrelated tests with ~/ since it's just plain text and may be valid for end-users.

#427

MVC PR to add app-relative URL resolution: https://github.com/aspnet/Mvc/pull/2875

### CODE REVIEW

**Comment by ajaybhargavb** [comment]
This test `AttributeBlockChunkGeneratorTest.GetHashCode_ReturnsSameValue_WhenEqual` is failing.

**Comment by NTaylorMullen** [comment]
@ajaybhargavb  https://github.com/aspnet/Razor/pull/465#issuecomment-125762926 :)

File: src/Microsoft.AspNet.Razor/CodeGenerators/GeneratedClassContext.cs [hunk 1]
```diff
@@ -136,8 +136,6 @@ public struct GeneratedClassContext
         public string WriteAttributeMethodName { get; set; }
         public string WriteAttributeToMethodName { get; set; }
 
-        [SuppressMessage("Microsoft.Design", "CA1056:UriPropertiesShouldNotBeStrings", Justification = "Property is not a URL property")]
-        public string ResolveUrlMethodName { get; set; }
```
Comment (dougbu) [inline src/Microsoft.AspNet.Razor/CodeGenerators/GeneratedClassContext.cs]: Cleanup this line as well

File: test/Microsoft.AspNet.Razor.Test/CodeGenerators/RazorChunkGeneratorTest.cs [hunk 2]
```diff
@@ -161,10 +161,7 @@ private Stream NormalizeNewLines(Stream inputStream)
                                                                    "DefineSection",
                                                                    "Instrumentation.BeginContext",
                                                                    "Instrumentation.EndContext",
-                                                                   new GeneratedTagHelperContext())
-            {
-                ResolveUrlMethodName = "Href"
-            };
+                                                                   new GeneratedTagHelperContext());
```
Comment (ajaybhargavb) [inline test/Microsoft.AspNet.Razor.Test/CodeGenerators/RazorChunkGeneratorTest.cs]: Is this what Ryan calls a #YOLO break?
And maybe add a blank line below this.

**Comment by ajaybhargavb** [comment]
:shipit:

File: test/Microsoft.AspNet.Razor.Test/Parser/Html/HtmlAttributeTest.cs [hunk 3]
```diff
@@ -122,30 +122,6 @@ public void MultiValueExpressionAttribute()
         }
 
         [Fact]
-        public void VirtualPathAttributesWorkWithConditionalAttributes()
-        {
-            ParseBlockTest("<a href='@foo ~/Foo/Bar' />",
```
Comment (dougbu) [inline test/Microsoft.AspNet.Razor.Test/Parser/Html/HtmlAttributeTest.cs]: Would prefer we keep one of these, showing the straightforward set of `Span`s and `Block`s generated. Helps to show `~` remains not special at this level.

**Comment by dougbu** [comment]
:shipit: with one restored (and simplified) test

File: test/Microsoft.AspNet.Razor.Test/CodeGenerators/RazorChunkGeneratorTest.cs (hunk 2 above)
Comment (dougbu) [inline test/Microsoft.AspNet.Razor.Test/CodeGenerators/RazorChunkGeneratorTest.cs]: @ajaybhargavb I'm curious what you're objecting to on this line?

Comment (ajaybhargavb) [inline test/Microsoft.AspNet.Razor.Test/CodeGenerators/RazorChunkGeneratorTest.cs]: I thought we are supposed to use this styling
```C#
new GeneratedClassContext(
    GeneratedClassContext.DefaultExecuteMethodName, 
    GeneratedClassContext.DefaultWriteMethodName, 
    GeneratedClassContext.DefaultWriteLiteralMethodName, 
    "WriteTo" )
```
over
```C#
new GeneratedClassContext(GeneratedClassContext.DefaultExecuteMethodName, 
                          GeneratedClassContext.DefaultWriteMethodName, 
                          GeneratedClassContext.DefaultWriteLiteralMethodName, 
                          "WriteTo" )
```

Comment (NTaylorMullen) [inline test/Microsoft.AspNet.Razor.Test/CodeGenerators/RazorChunkGeneratorTest.cs]: We do now. It's just old formatting :smile:. Will update.

Comment (ajaybhargavb) [inline test/Microsoft.AspNet.Razor.Test/CodeGenerators/RazorChunkGeneratorTest.cs]: :+1: Got it.

**Pull request merged**